### DISCUSSION TITLE
In Rust, how to traverse all AST nodes and print node type and span?

### DISCUSSION THREAD

**Discussion by edupsousa** [swc-project/swc#6766] [Q&A] [answered]
Hi, there's simple way (without the need to implement visit_xxx for every type of node) to traverse an entire AST and print the node type and span for each node? I found a previous issue (#501) on this topic, but the example requires a generic version of the Fold/Visit trait that seems outdated.

Thank you.

**Comment by kdy1** [comment, marked as answer]
No. You should use visitor
